### DISCUSSION TITLE
MySQL Relations not working

### DISCUSSION THREAD

**Discussion by JensBond** [Budibase/budibase#4624] [Help] [answered]
Hi,

I connected BudiBase to a MySQL Database. I created the relations in Budibase. But the Dropdown for the Datasource on the DataProvider element does not show the relations. So I can only select the Main Tables there.

What can I do?

Thank you.

**Comment by mjashanks** [comment, marked as answer]
Hi @JensBond 

I just tested this, and it is working ok for me. Possibly the mistake that you are making, is that those relationships only appear when the DataProvider is in the context of the parent relationship... I'll try to explain..

I have the DB model
- Customers table
- Invoices table 
- One customer has many invoices

Then my component structure looks like
```
> Data Provider : Customer, filtered by ID
  > Repeater
     > Data Provider : Invoices <----- Relationship: Repeater.Customer.Invoices
         > Repeater + other components
```

So, the "Relationships" are only available when inside the "Parent" customer record (repeater).

**Reply by JensBond** [reply]
Thank you that solved the problem
